### PR TITLE
[post] Pub-Sub Mesaging

### PR DESCRIPTION
**Pull request by roelbondoc**
Todo:
- [x] diagrams as mentioned

### CODE REVIEW

**Comment by roelbondoc** [comment]
@sammyd nope! I'm going through my edits tonight, will let you know when they are done.

**Comment by roelbondoc** [comment]
@sammyd I've gone through and made some changes. I also updated my diagrams. Let me know what you think.

Also, I'm not 100% attached to my diagrams, if you want to suggest any changes or refinements feel free.

Thanks!

File: _posts/2018-12-14-working-with-pub-sub-messaging.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+We can now increase the amount of notifications we can handle and allow us to serve real users without having to scale horizontally. Another added benefit of this is we can fine tune how many worker process we need and even scale it based on the load on the queue. This can all be done without affecting *carolus*' ability to serve our users.
+
+Working with Amazon SNS has been a great experience in learning about how pub-sub systems work. The two key take aways that I can generalize are as follows:
+
+* Ensure your messages are idempotent. This makes both the publishing and subscribing implementations simpler, and in turn makes debugging issues easier.
+* Defer any action or logic away from the processing of messages in your subscribers. Not only will you be able to handle more traffic, but it will also make testing simpler by being more modular.
+
+Implementing a message bus is integral in properly organizing the flow of data in your service oriented architecture. Just keep in mind that it is not an end all solution, but it can solve a lot of your problems when used carefully. If you want to learn more about our experiences or just want to chat, feel free to hit me up on Twitter!
```
Comment (sammyd) [inline _posts/2018-12-14-working-with-pub-sub-messaging.md]: Add a link to your twitter handle

**Comment by sammyd** [comment]
Hey @roelbondoc this is a looking a lot smoother—nice work. I think the diagrams are fine. The first one is maybe a little confusing because it could imply that there are always the same number or publishers as topics and as subscribers—i.e. it doesn't convey the fact it can be a conceptual spaghetti mess.

I've asked @Manda94 to take a look over this post by Wednesday, so it'll definitely be ready to publish this week.

**Comment by luke-freeman** [comment]
@roelbondoc I've added a hero for your post

**Pull request merged**